core/src/main/java/com/yahoo/imapnio/async/data/ExtendedSearchSequence.java

```diff
@@ -0,0 +1,192 @@
+package com.yahoo.imapnio.async.data;
+
+import com.sun.mail.iap.Argument;
+import com.sun.mail.imap.protocol.SearchSequence;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.mail.search.AndTerm;
+import javax.mail.search.FlagTerm;
+import javax.mail.search.NotTerm;
+import javax.mail.search.OrTerm;
+import javax.mail.search.SearchException;
+import javax.mail.search.SearchTerm;
+
+import java.io.IOException;
+
+/**
+ * This class extends the search sequence for modification sequence with the optional fields entry
+ * name and type defined in https://tools.ietf.org/html/rfc7162#section-3.1.5.
+ */
+public class ExtendedSearchSequence extends SearchSequence {
+
+    /**
+     * Generate the IMAP search sequence for the given search expression.
+     *
+     * @param term the search term
+     * @return the IMAP search sequence argument
+     * @throws SearchException will not throw
+     * @throws IOException will not throw
+     */
+    public Argument generateSequence(@Nonnull final SearchTerm term) throws IOException, SearchException {
+        return generateSequence(term, null);
+    }
+
+    /**
+     * Generate the IMAP search sequence for the given search expression.
+     *
+     * @param term the search term
+     * @param charset the character set
+     * @return the IMAP search sequence argument
+     * @throws SearchException will not throw
+     * @throws IOException will not throw
+     */
+    @Override
+    public Argument generateSequence(@Nonnull final SearchTerm term, @Nullable final String charset)
+            throws SearchException, IOException {
+        if (term instanceof ExtendedModifiedSinceTerm) {
+            return modifiedSince((ExtendedModifiedSinceTerm) term);
+        } else if (term instanceof AndTerm) {
+            return and((AndTerm) term, charset);
+        } else if (term instanceof OrTerm) {
+            return or((OrTerm) term, charset);
+        } else if (term instanceof NotTerm) {
+            return not((NotTerm) term, charset);
+        } else {
+            return super.generateSequence(term, charset);
+        }
+    }
+
+    /**
```

Comment:
No need to copy the code from com.sun.mail.imap.protocol.SearchSequence, just call super.generateSequence() from above method.

Reply:
If it calls super.generateSequence and there is another Or Search Term, it will use or() function from SearchSequence class. Eventually, it will reach the ExtendedModifiedSinceTerm and throw exception since sun mail imap doesn't support the optional entry name and entry type fields.